nucleus/__init__.py

```diff
@@ -518,6 +519,73 @@ def create_model(
             client=self,
         )
 
+    def create_launch_model(
+        self,
+        name: str,
+        reference_id: str,
+        bundle_args: Dict[str, any],
+        metadata: Optional[Dict] = None,
+    ) -> Model:
+        launch_client = LaunchClient(api_key=self.api_key)
+
+        model_exists = any(model.name == name for model in self.list_models())
+        bundle_exists = any(
+            bundle.name == name + "-nucleus-autogen"
+            for bundle in launch_client.list_model_bundles()
+        )
+
+        if bundle_exists or model_exists:
+            raise ModelCreationError(
+                "Bundle with the given name already exists, please try a different name"
+            )
+
+        kwargs = {
+            "model_bundle_name": name + "-nucleus-autogen",
+            **bundle_args,
+        }
+
+        bundle = launch_client.create_model_bundle(**kwargs)
+        return self.create_model(
+            name,
+            reference_id,
+            metadata,
+            bundle.name,
+        )
+
+    def create_launch_model_from_dir(
+        self,
+        name: str,
+        reference_id: str,
+        bundle_from_dir_args: Dict[str, any],
+        metadata: Optional[Dict] = None,
+    ) -> Model:
+        launch_client = LaunchClient(api_key=self.api_key)
+
+        model_exists = any(model.name == name for model in self.list_models())
+        bundle_exists = any(
+            bundle.name == name + "-nucleus-autogen"
+            for bundle in launch_client.list_model_bundles()
+        )
+
+        if bundle_exists or model_exists:
+            raise ModelCreationError(
+                "Bundle with the given name already exists, please try a different name"
+            )
+
+        kwargs = {
+            "model_bundle_name": name + "-nucleus-autogen",
```

Comment:
@seanshi-scale something i do wonder is how we should name this bundle... does a suffix make sense or should we just make it named the name passed in in general?

Reply:
hmm this seems like an "api" question? I feel like I don't have too strong of an opinion but ideally something that says "created from nucleus" e.g. something in `bundle_metadata` (lol) or in the name feels like a good idea. (Also I feel like if clients have a deeper integration, adding a suffix might cause weird behavior)

Reply:
Generally speaking, it doesn't matter how you identify the Nucleus-associated objects, as long as your client library functions can identify them again (i.e. you define the inverse operation). That said, all other things held equal, it'd probably be better to do this in a more structured way, which means that metadata (even though metadata itself is unstructured) is better than just a naming convention.

Reply:
Another thought, assuming we put things into metadata: we may want to reserve a convention that metadata fields that have a certain prefix (e.g. `_` or `__` are "internal", aka they're used by us for bookkeeping for things like Nucleus identification).

Reply:
re: nucleus -> launch querying, we have a column in the nucleus database for `bundle_id`

launch -> nucleus querying _i think_ would not be needed, since any bundle created thru the nucleus client has an attached model

i like the __ ideal for internal too hmm

Reply:
also i'd prefer not stuffing things into internal metadata if possible, we are using relational dbs for a reason, if we need it can just make a new column called "generated_by" and have it contain the internal product that made the bundle / endpoint

Reply:
Ok chatted a bit about this offline, but: it sounds like we don't need to do any Launch -> Nucleus querying (i.e. we do not need to know if a given Launch bundle has a Nucleus model; instead, the existing Nucleus model's `bundle_id`/`bundle_name` column is sufficient). So that means that any sort of metadata isn't a factor here 👌🏻 

All that you're really trying to accomplish here is give the endpoint _some_ name; adding an autogenerated suffix is purely a convenient visual marker (if we needed more structured querying, we could use some sort of other column or metadata - but as per the above discussion, we don't even need that). I think what @seanshi-scale is pointing out is that there may be an edge case where the customer has some workflow that assumes a certain suffix, and there might be subtle conflicts if that suffix matches ours. I think the way to go about this is to just claim a certain suffix convention for ourselves *as part of the public API*, e.g. `-scale-autogen`. This should discourage people from adopting the same suffix, but even if they do, catastrophic results seem unlikely.

tl;dr - suffix sounds fine, just pick something you're comfortable committing to as part of the public API, with the understanding it's a "soft" requirement.

Reply:
I think you're implicitly assuming that the user has an expectation that `<name>` is what's used for both the bundle and the endpoint. I don't think we need to worry about this, because:
1. We can publish this suffix convention as part of our API spec.
2. Orthogonally, also as part of the API docs, we should show users that they should `get` the resulting endpoint and its name from this function, instead of assuming a particular pattern.